src/lib/plugins/placeBlock.js

```diff
@@ -89,16 +89,15 @@ module.exports.player = function (player, serv, { version }) {
       reference: referencePosition,
       playSound: true,
       sound: 'dig.' + (materialToSound[blocks[id].material] || 'stone')
-    }, ({ position, playSound, sound, id, damage }) => {
+    }, ({ position, playSound, sound, id, damage, heldItem }) => {
       if (playSound) {
         serv.playSound(sound, player.world, placedPosition.clone().add(new Vec3(0.5, 0.5, 0.5)), {
           pitch: 0.8
         })
       }
 
       if (player.gameMode === 0) { player.inventory.slots[36 + player.heldItemSlot]-- }
-
-      player.setBlock(position, id, damage)
+      player.setBlock(position, id, damage, heldItem.name)
```

Comment:
You should retrieve the stateId from the name in placeBlock and only give the stateId to setBlock()